src/controller/scan/base_controller.go

```diff
@@ -313,6 +313,24 @@ func (bc *basicController) Scan(ctx context.Context, artifact *ar.Artifact, opti
 	return nil
 }
 
+// Stop scan job of a given artifact
+func (bc *basicController) Stop(ctx context.Context, artifact *ar.Artifact) error {
+	if artifact == nil {
+		return errors.New("nil artifact to stop scan")
+	}
+	query := q.New(q.KeyWords{"extra_attrs.artifact.digest": artifact.Digest})
+	executions, err := bc.execMgr.List(ctx, query)
+	if err != nil {
+		return err
+	}
+	if len(executions) == 0 {
+		message := fmt.Sprintf("no scan job for artifact digest=%v", artifact.Digest)
+		return errors.BadRequestError(nil).WithMessage(message)
+	}
+	execution := executions[0]
+	return bc.execMgr.Stop(ctx, execution.ID)
```

Comment:
the execMgr.stop actually will not raise the error on cancelling a task, we need to think about how to report the errors to end user.

Reply:
> the execMgr.stop actually will not raise the error on cancelling a task, we need to think about how to report the errors to end user.

Harbor currently raises an error when cancelling a task only if cancelling the last task (out of a list of tasks managed by execMgr) errors out.

Reply:
if you trace the code, it eventlly can task.Stop() to stop scan. But task manager is just log the error instead of raising it, how does user know the failures from the API response?

```
for _, task := range tasks {
		if err = e.taskMgr.Stop(ctx, task.ID); err != nil {
			log.Errorf("failed to stop task %d: %v", task.ID, err)
			continue
		}
}
```

Reply:
> if you trace the code, it eventlly calles task.Stop() to stop scan. But task manager is just log the error instead of raising it, how does user know the failures from the API response?
> 
> ```
> for _, task := range tasks {
> 		if err = e.taskMgr.Stop(ctx, task.ID); err != nil {
> 			log.Errorf("failed to stop task %d: %v", task.ID, err)
> 			continue
> 		}
> }
> ```
 if it errors out when cancelling the last task, it will report this error,  there is a `return err` after this for loop. for those error out in the middle of a list of tasks, it will only log the error. whether reports error depends on cancelling the last task.